site/en/blog/migrating-to-typescript/index.md

```diff
@@ -0,0 +1,166 @@
+
+[During our migration to JavaScript modules](/blog/migrating-to-js-modules/), we had already discovered that Closure Compiler was uncovering more issues than it had previously. Moving to a standard module format had increased Closure’s ability to understand our codebase and therefore increased the effectiveness of type checkers.
+However, the TypeScript team was using a baseline version of DevTools that predated the JavaScript modules migration.
+Therefore, we had to figure out if the migration to JavaScript modules had also reduced the amount of errors the TypeScript compiler would catch.
+
+## Evaluating TypeScript
+
+DevTools has existed for over decade, in which it has grown to a considerably sized and feature-rich web application.
+At the time of writing this blog post, DevTools contains approximately 150,000 lines of first-party JavaScript code.
+When we ran the TypeScript compiler on our source code, the sheer volume of errors was overwhelming.
+We were able to figure out that while the TypeScript compiler was emitting fewer errors related to code resolution (~2,000 errors), there were still a further 6,000 errors present in our codebase related to type compatibility.
+
+This showed that while TypeScript was able to understand how to resolve types, it found a significant amount of type incompatibilities in our codebase.
+A manual analysis of these errors had shown that TypeScript was (most of the time) correct.
+The reason TypeScript was able to detect these and Closure was not was because often the Closure compiler would deduce a type to be an `Any`, whereas TypeScript would perform type inference based on assignments and infer a more accurate type.
+As such, **TypeScript was indeed better at understanding the structure of our objects and discovered problematic usages**.
+
+One important catch to this, is that the usage of the Closure compiler in DevTools included the frequent usage of `@unrestricted`.
+Annotating a class with `@unrestricted` effectively turns off the strict property checks of the Closure compiler for that specific class, which means that a developer can augment a class definition at will without type safety.
+We couldn't find any historical context as to why the usage of `@unrestricted` was prevalent in the DevTools codebase, but it had resulted in running Closure compiler in a less safe mode of operation for large portions of the codebase.
+
+A cross-analysis of our regressions with the type errors TypeScript discovered also showed an overlap, which led us to believe TypeScript could have prevented these issues (provided that the types themselves were correct).
+
+## Making `any` call
```

Comment:
Unfortunately the backticks are not really visible, but that's more of a stylistic comment on the rendering on the website rather than the contents of this post I suppose.

Reply:
Yea, will talk to the team on the styling to make it more visible.